network/wsNetwork.go

```diff
@@ -1956,6 +1978,57 @@ func (wn *WebsocketNetwork) prioWeightRefresh() {
 	}
 }
 
+func primaryNetwork(network protocol.NetworkID) string {
+	return fmt.Sprintf("algorand-%s.network", network)
+}
+
+func secondaryNetwork(network protocol.NetworkID) string {
+	return fmt.Sprintf("algorand-%s.net", network)
+}
+
+var networkIDToPrimarySRVBaseURIRegex = map[protocol.NetworkID]*regexp.Regexp{
+	config.Mainnet: regexp.MustCompile(primaryNetwork(config.Mainnet)),
+	config.Testnet: regexp.MustCompile(primaryNetwork(config.Testnet)),
+}
+
+var networkIDToSecondarySRVBaseURIRegex = map[protocol.NetworkID]*regexp.Regexp{
+	config.Mainnet: regexp.MustCompile(secondaryNetwork(config.Mainnet)),
+	config.Testnet: regexp.MustCompile(secondaryNetwork(config.Testnet)),
+}
+
+func (wn *WebsocketNetwork) mergePrimarySecondaryRelayAddressLists(network protocol.NetworkID,
+	primaryRelayAddresses []string, secondaryRelayAddresses []string) (dedupedRelayAddresses []string) {
+	// Initialize w/ worst case capacity
+	//dedupedRelayAddresses = make([]string, 0, len(primaryRelayAddresses)+len(secondaryRelayAddresses))
+	//copy(dedupedRelayAddresses, primaryRelayAddresses)
+
+	var relayAddressPrefixToValue = make(map[string]string, 2*len(primaryRelayAddresses))
+	var prgx = networkIDToPrimarySRVBaseURIRegex[network]
+
+	for _, pra := range primaryRelayAddresses {
+		var normalizedPra = strings.ToLower(pra)
+		relayAddressPrefixToValue[prgx.ReplaceAllString(normalizedPra, "")] = normalizedPra
+	}
+
+	var srgx = networkIDToSecondarySRVBaseURIRegex[network]
+
+	for _, sra := range secondaryRelayAddresses {
+		var normalizedSra = strings.ToLower(sra)
+		var pfxKey = srgx.ReplaceAllString(normalizedSra, "")
+		// Add to our address map if prefix does not exist, otherwise take no action (already covered by primary)
+		if relayAddressPrefixToValue[pfxKey] == "" {
```

Comment:
This only works because an empty string isn't a valid value, use the second result from a map access to check if the map contains the key:
```suggestion
		// Add to our address map if prefix does not exist, otherwise take no action (already covered by primary)
		if _, ok := relayAddressPrefixToValue[pfxKey] == !ok {
```

Reply:
Fixed while writing the tests exercising this :)